pkg/actions/cluster/delete.go

```diff
@@ -0,0 +1,115 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+	"github.com/weaveworks/eksctl/pkg/fargate"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/elb"
+	"github.com/weaveworks/eksctl/pkg/kubernetes"
+	ssh "github.com/weaveworks/eksctl/pkg/ssh/client"
+	"github.com/weaveworks/eksctl/pkg/utils/kubeconfig"
+
+	"github.com/kris-nova/logger"
+)
+
+func deleteSharedResources(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, clientSet kubernetes.Interface, waitTimeout time.Duration) error {
+	clusterOperable, _ := ctl.CanOperate(cfg)
+	if clusterOperable {
+		if err := deleteFargateProfiles(cfg.Metadata, waitTimeout, ctl); err != nil {
+			return err
+		}
+	}
+
+	stackManager := ctl.NewStackManager(cfg)
+
+	if hasDeprecatedStacks, err := deleteDeprecatedStacks(stackManager); hasDeprecatedStacks {
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
do we not need to keep going here?

Reply:
good point, again copied from the original implementation. Perhaps the point here is that if the cluster has deprecated stacks, the rest of the logic doesn't apply? Thoughts @cPu1 @michaelbeaumont

Reply:
yeh i didn't know off the top of my head if the subsequent skipped steps in this case were needed or not 🤷‍♀️